overlord/exportstate/manifest.go

```diff
@@ -0,0 +1,203 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package exportstate
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+)
+
+// Manifest describes content content exported to snaps or the host.
+type Manifest struct {
+	ExportedName    string `json:"exported-name"`
+	ExportedVersion string `json:"exported-version"`
+
+	SnapInstanceName string `json:"snap-instance-name"`
+	SnapRevision     string `json:"snap-revision"`
```

Comment:
this should be a snap.Revision I think

Reply:
Sure, it can be if we keep `ExportedVersion` as a string, to accommodate things like `host` and `core_42`

Reply:
I'll do this change next.

Reply:
This is now done.

Reply:
This is now done.